tracer/src/Datadog.Trace/Processors/RedisObfuscationUtil.cs

```diff
@@ -0,0 +1,948 @@
+
+            var firstLetter = char.ToUpperInvariant(query[startIndex]);
+
+            if (firstLetter == 'C')
+            {
+                if (char.ToUpperInvariant(query[startIndex + 1]) == 'L')
+                {
+                    return (endIndex == startIndex + 6
+                         && char.ToUpperInvariant(query[startIndex + 2]) == 'I'
+                         && char.ToUpperInvariant(query[startIndex + 3]) == 'E'
+                         && char.ToUpperInvariant(query[startIndex + 4]) == 'N'
+                         && char.ToUpperInvariant(query[startIndex + 5]) == 'T')
+                        || (endIndex == startIndex + 7
+                         && char.ToUpperInvariant(query[startIndex + 2]) == 'U'
+                         && char.ToUpperInvariant(query[startIndex + 3]) == 'S'
+                         && char.ToUpperInvariant(query[startIndex + 4]) == 'T'
+                         && char.ToUpperInvariant(query[startIndex + 5]) == 'E'
+                         && char.ToUpperInvariant(query[startIndex + 6]) == 'R');
+                }
+                else if (char.ToUpperInvariant(query[startIndex + 1]) == 'O')
+                {
+                    return (endIndex == startIndex + 6
+                         && char.ToUpperInvariant(query[startIndex + 2]) == 'N'
+                         && char.ToUpperInvariant(query[startIndex + 3]) == 'F'
+                         && char.ToUpperInvariant(query[startIndex + 4]) == 'I'
+                         && char.ToUpperInvariant(query[startIndex + 5]) == 'G')
+                        || (endIndex == startIndex + 7
+                         && char.ToUpperInvariant(query[startIndex + 2]) == 'M'
+                         && char.ToUpperInvariant(query[startIndex + 3]) == 'M'
+                         && char.ToUpperInvariant(query[startIndex + 4]) == 'A'
+                         && char.ToUpperInvariant(query[startIndex + 5]) == 'N'
+                         && char.ToUpperInvariant(query[startIndex + 6]) == 'D');
+                }
+            }
+            else if (firstLetter == 'D')
+            {
+                return (endIndex == startIndex + 5)
+                    && char.ToUpperInvariant(query[startIndex + 1]) == 'E'
+                    && char.ToUpperInvariant(query[startIndex + 2]) == 'B'
+                    && char.ToUpperInvariant(query[startIndex + 3]) == 'U'
+                    && char.ToUpperInvariant(query[startIndex + 4]) == 'G';
+            }
+            else if (firstLetter == 'S')
+            {
+                return (endIndex == startIndex + 6)
+                    && char.ToUpperInvariant(query[startIndex + 1]) == 'C'
+                    && char.ToUpperInvariant(query[startIndex + 2]) == 'R'
+                    && char.ToUpperInvariant(query[startIndex + 3]) == 'I'
+                    && char.ToUpperInvariant(query[startIndex + 4]) == 'P'
+                    && char.ToUpperInvariant(query[startIndex + 5]) == 'T';
+            }
```

Comment:
nit: I think there's room for improvements here.

Reply:
Open to suggestions :D